lib/reek/smells/nested_iterators.rb

```diff
@@ -70,8 +70,10 @@ def find_iters_for_iter_node(exp, depth)
       end
 
       def ignored_iterator?(exp)
-        name = exp.call.method_name.to_s
-        @ignore_iterators.any? { |pattern| /#{pattern}/ =~ name }
+        if exp.call.respond_to? :method_name
```

Comment:
It's probably better to implement `#methodname` for the case where it is now not implemented. To do this, you need to add a module `ZsuperNode` in `lib/reek/source/sexp_extensions` and implement `#method_name` there.

This will allow people to mark `super` as an ignored iterator, and avoids the nasty `respond_to?`.